src/main/java/interface_adapter/execute_buy/ExecuteBuyController.java

```diff
@@ -15,11 +15,15 @@ public ExecuteBuyController(ExecuteBuyInputBoundary interactor) {
     }
 
     public void execute(String ticker, String quantity) {
-        final ExecuteBuyInputData data = new ExecuteBuyInputData(
+        try {
+            final ExecuteBuyInputData data = new ExecuteBuyInputData(
                 ClientSessionManager.Instance().getCredential(),
                 ticker,
                 Integer.parseInt(quantity));
 
-        interactor.execute(data);
+            interactor.execute(data);
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid quantity");
```

Comment:
It would be better to notify the user in some way, such as boardcasting a `DialogEvent`

Reply:
If quantity is not a valid number, the code simply prints "Invalid quantity". But, it does not provide feedback to the caller about what went wrong. Instead, you should either throw an exception or return an error message to the caller.